error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/NestedOptionals.java

```diff
@@ -41,8 +41,11 @@ public NestedOptionals() {}
 
   @Override
   public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState state) {
-    return state.getTypes().isSubtype(ASTHelpers.getType(tree), OPTIONAL_OF_OPTIONAL.get(state))
-        ? describeMatch(tree)
-        : Description.NO_MATCH;
+    Type stateType = OPTIONAL_OF_OPTIONAL.get(state);
+    if (stateType != null && state.getTypes().isSubtype(ASTHelpers.getType(tree), stateType)) {
+      return describeMatch(tree);
+    } else {
+      return Description.NO_MATCH;
+    }
```

Comment:
Alright, so I guess that the issue happens if the `Optional` class isn't loaded during compilation at all. That realization also lead to [this code](https://github.com/PicnicSupermarket/error-prone-support/blob/96114235c50217b09cb2e5dfd77b36d803ae614c/error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/util/ThirdPartyLibrary.java#L77-L85).

Reply:
We do have a preference for an early return with `Description.NO_MATCH`:
 ```suggestion
    Type type = OPTIONAL_OF_OPTIONAL.get(state);
    if (type == null || !state.getTypes().isSubtype(ASTHelpers.getType(tree), type)) {
      return Description.NO_MATCH;
    }

    return describeMatch(tree);
```